### PR TITLE
implement psbt_v0_to_v2

### PR DESCRIPTION
**Pull request by instagibbs**
Not tested. Needed for CLN upstream since our plan is to handle all PSBTs internally as V2, then export as V0 as required.

### CODE REVIEW

**Comment by instagibbs** [comment]
@achow101 would appreciate an eyeball to see if this is on the right track

File: src/psbt.c [hunk 1]
```diff
@@ -3662,9 +3662,48 @@ static int psbt_build_tx(const struct wally_psbt *psbt, struct wally_tx **tx,
...
+        return ret;
+
+    /* We explicitly do not support PSETv0 */
+    if (is_pset)
+        return WALLY_EINVAL;
+
+    psbt->version = PSBT_2;
+    psbt->tx_version = psbt->tx->version;
+    psbt->fallback_locktime = psbt->tx->locktime;
+    psbt->has_fallback_locktime = false;
```
Comment (achow101) [inline src/psbt.c:3682]: false?

**Review by achow101 — COMMENTED** [review]
I think `psbt->num_inputs` and `psbt->num_outputs` need to be set?

**Comment by instagibbs** [comment]
I think that's unneeded as per "psbt_set_global_tx" which sets those. Waiting on a set of test round trip vectors from jon before continuing :+1:

**Comment by jgriffiths** [comment]
@instagibbs see https://github.com/ElementsProject/libwally-core/commit/8af7892ad069acfb77dfb9ed2119643dc83b9ca1 for an example test

File: src/psbt.c [hunk 2]
```diff
@@ -3662,9 +3662,48 @@ static int psbt_build_tx(const struct wally_psbt *psbt, struct wally_tx **tx,
...
+            return ret;
+    }
+
+    ret = wally_tx_free(psbt->tx);
+    psbt->tx = NULL;
+
+    return ret;
 }
 
 static int psbt_v2_to_v0(struct wally_psbt *psbt)
```
Comment (instagibbs) [inline src/psbt.c:3709]: this routine needs to set `psbt->tx = tx` as well

**Comment by instagibbs** [comment]
Wrote up some more tests in https://github.com/instagibbs/libwally-core/commit/05ec12c1cf36f18307610ce45202dabddd826c81

@jgriffiths I can push a fix for the global tx issue and more tests in this PR if you like?

**Comment by instagibbs** [comment]
closing as #364 implemented all changes

**Pull request closed without merging**